pdf/model/colorspace.go

```diff
@@ -509,14 +509,17 @@ func (this *PdfColorspaceDeviceCMYK) ColorToRGB(color PdfColor) (PdfColor, error
 	y := cmyk.Y()
 	k := cmyk.K()
 
-	c = c*(1-k) + k
-	m = m*(1-k) + k
-	y = y*(1-k) + k
-
+	// Peter Williams. I need a cmyk->rgb conversion that gives r!=g or g!=b when c!=m or m!=y
```

Comment:
I defined the dummy encodings to make the following code work. Some encodings are only used for grayscale images and JPX RGB is typically used for color images. I needed some way of getting the encoding name for my switch. RLE seemed so easy that I implemented it before I realized I didn't need it.

```
    switch ximg.Filter.GetFilterName() {
    // TODO: Add JPEG2000 encoding/decoding. Until then we assume JPEG200 images are color
    case "JPXDecode":
        processedXObjects[string(*name)] = true
        colored = true
        return nil
    // These filters are only used with grayscale images
    case "CCITTDecode", "JBIG2Decode", "RunLengthDecode":
        return nil
    }
```
I don't need that cmyk->rgb conversion change any more so I can remove that code change.  Answer to general question about whether this change is application specific:

In printing, we care about cmyk because printer toners are cmyk. Color conversion is a complex topic but a short summary is: Device-independent cmyk is an unhelpful abstraction that exists in PDF files so we have to deal with it. I would expect that c!=m || c!=y || m!=y in a PDF could result in c, m and/or y toner being used to print that PDF file. Most printers will print rgb = 0,0,0 using only k toner. Therefore c!=m || c!=y || m!=y  should not be converted to rgb = 0,0,0